### ISSUE TITLE
Screen Savers broken function on 2+ monitor system in Ventura 13.3 beta (FB12023530)

### ISSUE THREAD

**Issue by xmddmx** [JohnCoates/Aerial#1286]
Hijacking the top post, there is currently a bug in version 13.3 of macOS, the bug is fixed in partially fixed in macOS 13.4 if you only have 2 monitors, and it will still randomly fail 5% of the time. 3 monitors always fails so far. More details here if you need a concise explanation : https://ko-fi.com/post/PSA-Multi-monitor-support-is-broken-in-macOS-Ven-E1E2JYZAL 

--

macOS Ventura 13.3 beta 2 seems to severely break screen savers on multimonitor systems.  

(This is not specific to Aerial, but describing it here in case others are wondering)

I just reported this issue to Apple as Feedback Case FB12023530:

To reproduce:
* on a M1 MacBook Air with a second monitor connected
* open System Preferences / Screen Saver and select the **Message** Screen Saver
* click the Preview button
* Expected behavior: screen saver should run on both monitors
* Actual behavior: screen saver runs on 2nd monitor only.
* Actual behavior: the screen’s rectangle seems to be offset by about 100 pixels (as if the NSView is not set to the actual screen coordinates)

Regression:
* this worked fine in 13.2
* this also affects third party screensavers such as **iScreensaver** and **Aerial** 
* this doesn’t affect all first party Apple screensavers, e.g. “Hello” and “Flurry” work fine, but “Message” does not

**Comment by glouel** [comment]
Thanks, can confirm, I also reported it. I did see we were initialized on both screen but didn't catch that no StartAnimation event was sent. I'll update my FB with a referal here.

**Comment by xmddmx** [comment]
In a prior version of macOS (can't remember if it was Monterey or Big Sur?) there was a bug where StartAnimation never fired, so I simply added a flag and a callback timer, e.g.

```
Init
  Set 1 second timer

StartAnimation
   bStartAnimationCalled=true

TimerCallback
   if bStartanimationCalled  == false
     StartAnimation()  // in case StartAnimation wasn't called, call it now
```

I wonder if this trick would help in Ventura?

**Comment by glouel** [comment]
I was having a second look, more precisely this part of your message : 

> Actual behavior: the screen’s rectangle seems to be offset by about 100 pixels (as if the NSView is not set to the actual screen coordinates)

I can't say I'm seeing this here, I get the correct coordinates passed in the init at least : 

> 2023-03-01 18:11:12.711 : avInit .saver (0.0, 0.0, 2560.0, 1440.0) p: false o: false
> 2023-03-01 18:11:12.859 : avInit .saver (2560.0, 0.0, 2560.0, 1440.0) p: false o: false

(this is one 5K monitor and one 2560x 1440 monitor, so retina factor 2 and 1 respectively) 

So at least here I can correctly match to the monitor informations. I don't seem to see an offset although maybe there's one and I didn't notice it with the videos. There's no black part anywhere at least. 

Regarding startAnimation, this was on a run where it launched correctly on both : 

2023-03-01 18:11:13.221 : <AerialView: 0x12a767eb0> startAnimation frame (0.0, 0.0, 2560.0, 1440.0) bounds (0.0, 0.0, 2560.0, 1440.0)
2023-03-01 18:11:13.383 : <AerialView: 0x12b8058c0> startAnimation frame (0.0, 0.0, 2560.0, 1440.0) bounds (0.0, 0.0, 2560.0, 1440.0)

This is however what I get with a more failed on first monitor launch : 

2023-03-01 18:21:07.325 : <AerialView: 0x13e80a610> startAnimation frame (0.0, 0.0, 2560.0, 1440.0) bounds (0.0, 0.0, 2560.0, 1440.0)

Only one call of startAnimation.

So yeah, there's a good chance your workaround could actually work, I'll see if I can give it a shot quickly, it's likely you got the root cause here.

**Comment by xmddmx** [comment]
I, too, am seeing the correct monitor coordinates in my logs, but then the playback on the second screen (which seems to actually be the first screen's process) is offset.

My monitor setup:
* M1 Macbook Air Built in display : Default scaling (frame = 0,0,  1440x900)
* External 24" monitor Default scaling (frame = 1440, -250, 1680x1050)

Arrangement:

<img width="520" alt="image" src="https://user-images.githubusercontent.com/107132172/222216225-03765f46-ca1a-4ccb-9b05-5d01da02fc7c.png">

However, if I mess around with monitor arrangement, it always looks the same, so I suspect it's not the monitor's relative locations, but perhaps their Sizes which is relevant?

Here's what I see when running.  The built-in screen is blank, and the 2nd monitor is offset:

<img width="765" alt="image" src="https://user-images.githubusercontent.com/107132172/222217156-03296c27-3c92-4ce1-bfab-412648200967.png">

**Comment by glouel** [comment]
Yeah I was going to say, this looks like the size of your main monitor running on the second, because mine are similar sized I didn't see that part. 

I'm wondering at what point the coordinates get wrecked, I'll try messing with resolution to dig a bit more.

**Comment by glouel** [comment]
Ok so, I tried your idea for a workaround but I think that the original issue that you noted is not allowing us to fix it ourselves : 


> 2023-03-01 18:40:39.752 : avInit .saver (0.0, 0.0, 2560.0, 1440.0) p: false o: false
> 2023-03-01 18:40:39.753 : <AerialView: 0x13f13cd60> AerialView setup init (V3.2.4) preview: false

0x13f13cd60 : This is the init on main screen 

> 2023-03-01 18:40:39.921 : avInit .saver (2560.0, 0.0, 2560.0, 1440.0) p: false o: false
> 2023-03-01 18:40:39.922 : <AerialView: 0x139922a60> AerialView setup init (V3.2.4) preview: false

0x139922a60 : This is the second screen, correct coordinates here

Now, this is the startAnimation that goes through : 
 
> 2023-03-01 18:40:40.274 : <AerialView: 0x13f13cd60> startAnimation frame (0.0, 0.0, 2560.0, 1440.0) bounds (0.0, 0.0, 2560.0, 1440.0)

This is the init of the main screen. But we know it shows up on the secondary despite correct frame and bounds.

The workaround does kick in for the secondary monitor : 

> 2023-03-01 18:40:40.967 : forcing startAnimation (workaround Ventura 13.3beta
> 2023-03-01 18:40:40.968 : <AerialView: 0x139922a60> startAnimation frame (2560.0, 0.0, 2560.0, 1440.0) bounds (0.0, 0.0, 2560.0, 1440.0)

This does not do anything for me, it doesn't replace the first one for example.

Again coordinates are correct, but the view is not displayed where it should despite what it says. So I think you're correct here, for some reason in some cases, the main screen thread's NSView is mistakenly put on the second monitor. I'll try to double check by changing resolution.

**Comment by xmddmx** [comment]
yeah, it feels as if this is an OS bug.

If I run Console set to show Errors and Warnings, and test using apple's **Message** screensaver, I see errors which are suggestive of some sort of problem:

`error	09:49:11.053147-0800	CoreFoundation	ScreenSaverEngine	Attempt to load executable of a type that cannot be dynamically loaded for CFBundle 0x60000103a000 </System/Library/Frameworks/ScreenSaver.framework/PlugIns/Computer Name.appex> (executable, not loaded)
error	09:49:11.056956-0800	ExtensionFoundation	ScreenSaverEngine	Extension request contains input items but the extension point does not specify a set of allowed payload classes. The extension point's NSExtensionContext subclass must implement `+_allowedItemPayloadClasses`. This must return the set of allowed NSExtensionItem payload classes. In future, this request will fail with an error.
error	09:49:11.062119-0800	SkyLight	WindowServer	CGLayerKit called back with no connection
error	09:49:11.073596-0800	SkyLight	WindowServer	CGLayerKit called back with no connection
error	09:49:11.064167-0800	pkd	pkd	could not fetch persona generation ID (Error Domain=PlugInKit Code=-1 "persona generation ID unavailable" UserInfo={NSLocalizedDescription=persona generation ID unavailable}), flushing cache
error	09:49:11.074242-0800	SkyLight	WindowServer	CGLayerKit called back with no connection
error	09:49:11.075145-0800	SkyLight	WindowServer	CGLayerKit called back with no connection
error	09:49:11.079260-0800	SkyLight	WindowServer	CGLayerKit called back with no connection
error	09:49:11.093786-0800	RunningBoard	runningboardd	memorystatus_control error: MEMORYSTATUS_CMD_CONVERT_MEMLIMIT_MB(-1) returned -1 22 (Invalid argument)
error	09:49:11.093914-0800	RunningBoard	runningboardd	memorystatus_control error: MEMORYSTATUS_CMD_CONVERT_MEMLIMIT_MB(0) returned -1 22 (Invalid argument)
error	09:49:11.100191-0800	SkyLight	WindowServer	CGLayerKit called back with no connection
error	09:49:11.107122-0800	AppKit	loginwindow	Warning: Window LUI2Window 0x143b1c060 ordered front from a non-active application and may order beneath the active application's windows.
error	09:49:11.107300-0800	AppKit	loginwindow	Warning: Window LUI2Window 0x143940890 ordered front from a non-active application and may order beneath the active application's windows.
error	09:49:11.107395-0800	AppKit	loginwindow	Warning: Window LUI2Window 0x143d1c110 ordered front from a non-active application and may order beneath the active application's windows.
error	09:49:11.107772-0800	AppKit	loginwindow	Warning: Window LUI2Window 0x143a0f780 ordered front from a non-active application and may order beneath the active application's windows.
error	09:49:11.108216-0800	AppKit	loginwindow	Warning: Window LUI2Window 0x143b1c060 ordered front from a non-active application and may order beneath the active application's windows.
error	09:49:11.109992-0800	launchservicesd	launchservicesd	Application App:"loginwindow" asn:0x0-2002 pid:568 refs=7 @ 0x1294098e0 tried to be brought forward, but isn't in fPermittedFrontApps ( ( "LSApplication:0x0-0x1bd1bd pid=6021 "Screen Saver"")), so denying.
error	09:49:11.110061-0800	SkyLight	WindowServer	0[SetFrontProcess]: [cps/setfront] Failed setting the front application to <private>, psn 0x0-0x2002, securitySessionID=0x186ad, err=-13066
`

**Comment by glouel** [comment]
Test with resolution changed to 1920 on secondary : 

> [id=7, width=2560, height=1440, bottomLeftFrame=(0.0, 0.0, 2560.0, 1440.0), topRightCorner=(2560.0, 1440.0), isMain=true, backingScaleFactor=2.0]
> [id=6, width=1920, height=1080, bottomLeftFrame=(2560.0, 360.0, 1920.0, 1080.0), topRightCorner=(4480.0, 1440.0), isMain=false, backingScaleFactor=1.0]

What I get 

> 2023-03-01 18:47:50.789 : avInit .saver (0.0, 0.0, 2560.0, 1440.0) p: false o: false
> 2023-03-01 18:47:50.789 : <AerialView: 0x13b033b30> AerialView setup init (V3.2.4) preview: false

0x13b033b30  : First screen 

> 2023-03-01 18:47:50.950 : avInit .saver (2560.0, 360.0, 1920.0, 1080.0) p: false o: false
> 2023-03-01 18:47:50.951 : <AerialView: 0x13b0dfe50> AerialView setup init (V3.2.4) preview: false

Second thread with correct coordinates + size.

> 2023-03-01 18:47:51.311 : <AerialView: 0x13b033b30> startAnimation frame (0.0, 0.0, 2560.0, 1440.0) bounds (0.0, 0.0, 2560.0, 1440.0)

This is the main screen frame that is started ! And visually I can confirm, in my case it shows blown up (missing top/right part) on the secondary screen. 

2023-03-01 18:47:51.994 : forcing startAnimation (workaround Ventura 13.3beta
2023-03-01 18:47:51.996 : <AerialView: 0x13b0dfe50> startAnimation frame (2560.0, 360.0, 1920.0, 1080.0) bounds (0.0, 0.0, 1920.0, 1080.0)

The secondary thread is force started but this is irrelevant. 

So again thanks for your input on this, I'll update the FB again (I reported it as FB12007471).

**Comment by xmddmx** [comment]
I just tested 13.3 beta 3 - no change, it's still malfunctioning.  :(

**Comment by DSBlackHeart** [comment]
Hello, I have this issue.  If someone needs help with testing or if I can help please let me know.
![Screenshot 2023-03-09 at 12 53 35 PM](https://user-images.githubusercontent.com/127451632/224173570-69524fee-3f33-4b12-90de-9d89f8addb2f.jpg)

**Comment by xmddmx** [comment]
I just tested 13.3 beta 4, and had very weird results.

Tried **Aerial**: worked
Tried a screensaver built using **iScreensaver**: did work.
Tried a different screensaver saver built using **iScreensaver**: did not work

Went back and tried the prior screensaver: did not work.

Went back and tried Aerial again.  did not work.

This seems reproducible:

* Restart in 13.3 beta 4
* Launch Aerial (using Hot Corner).  Result: works on 2 monitors
* Exit screensaver
* Launch Aerial (using Hot Corner).  Result: fails: only shows on 2nd monitor

Advice: If you test with 13.3 beta 4, don't just do a single test and say "good" - do some repeated tests.

Further testing:
* restarting restores normal operation (but only for 1 launch)
* logging out, back in: does not restore normal operation.

**Comment by glouel** [comment]
So I'm not sure if anything really changed, because it worked some of the time previously here, and maybe it works a bit more often but if I launch 10 times in a row I still get some working and some failings. It may work more often that before, but it's not as clear as you described at least here, failures and workings are still intertwined (just testing with Aerial multiple times in a row).

**Comment by xmddmx** [comment]
Good point, I think you are right.

I just did a bunch of tests in a row with both the Aerial and macOS Messages screensaver and you seem to be correct - it's failing pretty much randomly now, working about 20%-30% of the time.

X = fail
√ = works

Messages:  X X X X X √ X X X √ X X √ √√ √ X X √
Aerial: X √ X X X X X X X X X √ X X √ √ X √ √ X X X

I did not do this sort of repeated testing in 13.3 beta 3 or earlier, so I can't say if anything has changed in beta 4.

**Comment by DSBlackHeart** [comment]
Afternoon, I just updated to the Beta 4 & Aerial is still not working, I tried to start it by using my Hot Corner.  

I will see if a different SS has any better success. 


<img width="324" alt="Screenshot 2023-03-15 at 3 35 00 PM" src="https://user-images.githubusercontent.com/127451632/225459136-91100e32-6bb8-4d91-8e39-681e33e45a5d.png">

**Comment by DSBlackHeart** [comment]
Tried the following

Message - Did not do anything, tried preview & hot corner activation
Word Of Day - worked HC Activation
Ventura - worked HC Activation
Album Artwork - worked HC Activation
Monterey - worked HC Activation

**Comment by glouel** [comment]
> Good point, I think you are right.
> 
> pretty much randomly now, working about 20%-30% of the time.

Can confirm I have the same in b4

> I did not do this sort of repeated testing in 13.3 beta 3 or earlier, so I can't say if anything has changed in beta 4.

I didn't either, but I would have said it worked less frequently than that, maybe 15-20% ? At this point who knows. I updated my radar with the relevant info, I'm hopeful they are actively looking at it. Fingers crossed !

**Comment by DSBlackHeart** [comment]
You know My I-Mac woke me up the other day, it had a Panic Attack, I did not think about it till now, wonder if it had something to do with all this.....

Would the Diag Dump be some where ? If so would it help you ?

**Comment by DSBlackHeart** [comment]
Panic attack - it restarted the system

The stuff above i tested the screen is black but backlight is lit

**Comment by xmddmx** [comment]
I just tested the latest beta (which no longer says Beta, version = 13.3 (22E252) and the bug is still there.  Various websites are saying this version is the RC (Release Candidate) version.

Does anyone know a way to get some attention at Apple for this to escalate it?

**Comment by DSBlackHeart** [comment]
I still have the issue as well, I looked in the release notes and did not see any mention of a fix. The only problem with trying to get someone at Apple to do something about it is you have to go through about 10+ dumb people before you find someone who knows what is happening. 
The standard:
Delete the Prefs
reset the SMC
do a clean system install

Things you know 110% are not going to fix the problem.

**Comment by xmddmx** [comment]
Related discussion on Apple Developer Forums:  https://developer.apple.com/forums/thread/727168

**Comment by glouel** [comment]
@xmddmx @DSBlackHeart 

Just installed 13.4beta1 (released today) and thankfully, they fixed it! Let's hope they don't break it again and they release 13.4 quickly now ;)

**Comment by DSBlackHeart** [comment]
I am looking forward to trying out the update, I dont see it posted on my end yet but looking forward to the install. :)

**Comment by spawnofbill** [comment]
Unfortunately, updating to 13.4beta1 did not resolve this bug for me.

**Comment by glouel** [comment]
> Unfortunately, updating to 13.4beta1 did not resolve this bug for me. I am running a three monitor system on a Mac Studio

Do you have more info ? Can you replicate with the built-in Messages screensaver ? If so please make a bug report to Apple and mention FB12007471

**Comment by spawnofbill** [comment]
> > Unfortunately, updating to 13.4beta1 did not resolve this bug for me. I am running a three monitor system on a Mac Studio
> 
> Do you have more info ? Can you replicate with the built-in Messages screensaver ? If so please make a bug report to Apple and mention FB12007471

I cannot reproduce the issue with the built in Messages screensaver, it only happens with Aerial and other third party screensavers. I am seeing a single video playing on one of my secondary monitors but offset and smaller like shown above.
<img width="470" alt="Screenshot 2023-03-29 at 11 50 00 AM" src="https://user-images.githubusercontent.com/70985943/228598715-db373935-0399-4cdd-a979-bd70ca0fe743.png">

**Comment by xmddmx** [comment]
@spawnofbill  How are you launching the screensaver?  Letting it time out?  Using a Hot Corner?  Using the "Preview" button (which means the System Settings panel is open. These situations are different and relevant for how the legacyScreenSaver process works...

Also, if you are able, would you please try a demo screensaver from iScreensaver, such as:  https://iscreensaver.com/downloads/savers/InstallSpaceSaver.app.zip

**Comment by spawnofbill** [comment]
So interesting update, I just unplugged one of my monitors and Aerial worked perfectly fine. So something is wrong with how MacOS 13.4 handles more than two monitors now instead of just more than one.

**Comment by glouel** [comment]
I updated my feedback with Apple but please file one so they get what they need to reproduce.

**Comment by spawnofbill** [comment]
> I updated my feedback with Apple but please file one so they get what they need to reproduce.

Yup filed as FB12085443

**Comment by spawnofbill** [comment]
Oh this is fun. With *four* monitors it does this. 
![IMG_1561](https://user-images.githubusercontent.com/70985943/228656966-eddafae0-f161-4e5a-b93e-e34fb084e36a.jpeg)

**Comment by DSBlackHeart** [comment]
I know this feeling....

**Comment by DSBlackHeart** [comment]
I have checked several times but i dont see this update posted yet, I am still at 13.3 ( 22E252 )

**Comment by BrizzleRocker** [comment]
> I, too, am seeing the correct monitor coordinates in my logs, but then the playback on the second screen (which seems to actually be the first screen's process) is offset.
> 
> My monitor setup:
> 
> * M1 Macbook Air Built in display : Default scaling (frame = 0,0,  1440x900)
> * External 24" monitor Default scaling (frame = 1440, -250, 1680x1050)
> 
> Arrangement:
> 
> <img alt="image" width="520" src="https://user-images.githubusercontent.com/107132172/222216225-03765f46-ca1a-4ccb-9b05-5d01da02fc7c.png">
> 
> However, if I mess around with monitor arrangement, it always looks the same, so I suspect it's not the monitor's relative locations, but perhaps their Sizes which is relevant?
> 
> Here's what I see when running. The built-in screen is blank, and the 2nd monitor is offset:
> 
> <img alt="image" width="765" src="https://user-images.githubusercontent.com/107132172/222217156-03296c27-3c92-4ce1-bfab-412648200967.png">
> 
> Edit to add: in fact, I'm pretty sure that's the issue, the first screen is playing on the second screen, and since they aren't the same size, no wonder it doesn't fill the screen!

I'm seeing this exact same behaviour on Ventura 13.3 after the recent security update. I can confirm that Aerial was working just fine before. When Aerial is set to main display it doesn't show on the macbook, only the external screen.

**Comment by jbkkd** [comment]
Probably related, though not exactly the same - on a rotated screen, the video rotates wrong on the second screen and also cuts off (the video on the vertical screen is rotated 90 degrees, wrongly). On an M1 MBP
![WhatsApp Image 2023-03-30 at 13 13 47](https://user-images.githubusercontent.com/2523769/228804913-48035987-f7b6-400e-a55e-d6bfa531acad.jpeg)

**Comment by glouel** [comment]
@jbkkd the puzzling part to me on that one is that you have the text at the correct position at the bottom, so that means Aerial gets the correct stuff . It looks like a regression of an old bug though with the quicktime APIs not getting the rotation information, or getting wrong screen somehow. Weird.

**Comment by jbkkd** [comment]
Nope - I'm on 13.2.1, running Aerial 3.2.3

**Comment by DSBlackHeart** [comment]
Well did the update, NOT FIXED.
I guess my question is how hard is this to fix? It worked fine before and then it stopped, so what did they change? 
Let me guess..... Apple added some new, improved, great features that no one will really like or use. 
![IMG_2119](https://user-images.githubusercontent.com/127451632/228976069-d46d89b7-b11c-4a0d-8a8b-e80d64c3c81a.jpg)
![Screenshot 2023-03-30 at 3 05 24 PM](https://user-images.githubusercontent.com/127451632/228976078-28e96aad-4c08-45c9-97a9-252c9446917b.jpg)

**Comment by DSBlackHeart** [comment]
Just an open thought....

Has anyone tried to replicate this on the production release?

Ventura 13.3 use "terminal" to download the full installer and then install it in a disk partition to see if it still fails ?

Or did I just volunteer to have something to do on my day off ? LOL

**Comment by wblew** [comment]
This issue is occurring on my production Ventura 13.3. In fact here is Activity, note the %100 percent CPU as well.

![image](https://user-images.githubusercontent.com/40679887/229069794-1d0d628b-0849-40a6-a9ff-2e9aa3f75ad6.png)

**Comment by Fofer** [comment]
Yes, I came here to report the same.  I just added a 2nd 4K display to enjoy it with Ventura on my M1 Mac mini. Only once, did I see Aerial display with the video "spanning" across both screens, and now I can't get it to display that way again. It's only showing up on the left screen, not my primary.  The other screensavers I have installed are working fine.  Sad, as Aerial is, by far, the best screensaver in the universe!  Hopefully it can be addressed, otherwise I'll miss it greatly :(

**Comment by glouel** [comment]
@Fofer this is not an Aerial bug, it affects every single 3rd party screen saver and some 1st party screensaver (for example the Message one from Apple). As to why it worked once, the bug happen about 80-90% of the time, so you may see the odd time when everything works.

The bug is in macOS Ventura 13.3 specifically and 13.4 has a (partial for now) fix for the issue.

**Comment by JohnHales** [comment]
I installed MacOS 13.3 release version. I have 4 monitors (M1 Mackbook built in, 1 FHD and 2 4K monitors). Desktop Aerial background works great. Screensaver won't launch (screen just goes black). I have updated to the latest beta of both Companion and Aerial. Latest GA version had the same issue. I can view them via the app, but the screensaver won't start. Options via Settings doesn't work, but does via the Companion app. I tried having the video play on all monitors (independently - which I've used for years) as well as one by one individual windows and it won't start. I also found that while I can use Aerial on my main screen for desktop, on all secondary screens I just get the usual Wallpaper setting. I wondered after read the Docs and FAQs if it might be an advanced setting, so I disabled Brightness control and Vibrance (under Filters). I'm using 4K videos as the format. Overlays seem to make no difference. Looking forward to a resolution to this!!

**Comment by glouel** [comment]
> I installed MacOS 13.3 release version.  

Please see my message above. If you have access to macOS Betas, the bug is partially fixed in 13.4beta1, although it seems it doesn't work with 3 monitors (and likely 4 either). 

> Options via Settings doesn't work, but does via the Companion app. 

Yes there's something weird with System Settings, again. Closing and reopening System Settings seems to fix it.

**Comment by JohnHales** [comment]
> > I installed MacOS 13.3 release version.
 
> Please see my message above. If you have access to macOS Betas, the bug is partially fixed in 13.4beta1, although it seems it doesn't work with 3 monitors (and likely not 4 either).
>>  I figured. I'm not on the Apple Betas as I use it for work, but will update as soon as it is released.

> Options via Settings doesn't work, but does via the Companion app.
> 
> Yes there's something weird with System Settings, again. Closing and reopening System Settings seems to fix it.
>> I realized that it works the first time you open it after starting Settings, but if you close the Options window and try and reopen it, it doesn't open. The fi is to close and restart Settings.

**Comment by jiclark** [comment]
I can confirm that the issue is intermittent. I just opened System Settings and hit Preview, and Aerial showed the laptop-sized image on my 27-inch external, and nothing on the laptop screen. I then quit System Settings and initiated Aerial via a hot corner, and it behaved correctly. Exited Aerial, then re-initiated it via the hot corner, and it was borked again. Hope this might help troubleshoot the issue somehow?

[16-inch M1 Max MacBook Pro, running Ventura 13.3]

**Comment by glaphi** [comment]
Same here.

If you go to Experimental - Launch settings - Displays and select to only show on Main, it will only show on the external, if you chose external only - you get no screen saver. The laptop screen never shows the screen saver atm. For me, using hot corners didn't change anything

**Comment by glouel** [comment]
@glaphi @jiclark just in case you missed it in the amount of messages, the bug is in macOS 13.3 specifically, and is partially fixed in macOS 13.4beta1 (at least for 2 monitors, jury is still out on more monitors on latest beta).

See the message I added at  the very top of the post here for more details, as much as I'd like to, there's nothing I can do :(

**Comment by jiclark** [comment]
Thanks @glouel; I did see that above, but wanted you to know that it still works intermittently, in case that helps somehow. I absolutely adore this screensaver, so I can be patient while we wait for Apple to fix the damn bug...

**Comment by arorank** [comment]
I have noticed, being on macOS 13.3.1,

When screensavers are coming due to inactivity timeout they are working fine, but when forced using preview or alfred screen saver command they result in this issue.

**Comment by JohnHales** [comment]
It has not worked for me in any case.

Sent from John's iPad. Please excuse any typos.
________________________________

**Comment by kenmccann** [comment]
> I have noticed, being on macOS 13.3.1,
> 
> When screensavers are coming due to inactivity timeout they are working fine, but when forced using preview or alfred screen saver command they result in this issue.

Exactly the same for me on 13.3.1 - inactivity screensaver works fine.  Any attempt to manually start the screensaver (I use a hotkey which also locks the screen), will result in a broken screensaver.

**Comment by Daskarzine** [comment]
Mac OS Ventura 13.3.1 the screensaver on my primary 4k display stopped working.The screensaver on my secondary 1080 display still works. Glad it's not just me! 2023 MacBook Pro M2

**Comment by DSBlackHeart** [comment]
It has been a problem for a while, I miss my Aerial.... All 120+ GB of Aerial Videos....

**Comment by stonerl** [comment]
I install macOS 13.4 Beta 2, and it seems Apple has resolved this issue.

**Comment by xmddmx** [comment]
13.4beta 3 is out today.  Testing on a 2 monitor system (Macbook Air M1) shows what appears to be normal behavior: 20 consecutive HotCorner invocations of Aerial, 100% successful operation on both montiors.

I need to get 13.4 beta 3 installed on a system which can do 3+ monitors for futher testing.  Anyone out there have a 3+ monitor system who can test 13.4 beta 3?

**Comment by glouel** [comment]
@xmddmx do you allow your mac to control with mouse keyboard an ipad maybe ? I know that this scenario failed for me with 2 monitor +iPad.

I tried 2 monitors + iPad in extend mode (so kinda 3 monitors) and can confirm it *still* fails in that case.

**Comment by nickroberts** [comment]
3 monitors here...still only on half of one. Maybe some day! (DualUp + 38" Ultrawide + DualUp)

**Comment by spawnofbill** [comment]
Confirmed still broken with 3 monitors on Beta3

**Comment by mysoko** [comment]
After upgrading the operating system to 13.3.1, problems began to occur: the extended screen cannot be displayed in full screen (the laptop screen is black and not displayed). However, this situation does not happen every time, sometimes the display is good.

I am currently using version 3.2.4beta1 (other versions do not work).

In addition, here are my hardware configurations.
![image](https://user-images.githubusercontent.com/75619/235305616-897ce218-1091-4d78-a13c-01f4033934ef.png)
<img width="547" alt="image" src="https://user-images.githubusercontent.com/75619/235305650-d70955a2-f934-4229-9ddb-032428644f38.png">
<img width="313" alt="image" src="https://user-images.githubusercontent.com/75619/235305680-ef624f43-c31a-4e2a-a719-28d1559c113a.png">

**Comment by glouel** [comment]
Thanks for the report @mysoko. Per the other messages, the bug is in process of being fixed by Apple in macOS 13.4. For your case, only 2 monitors, the current 13.4 developer beta does fix the issue. 

To all, I did file again a new bug specific to the 3+ monitor situation (the one that is still not fixed in current betas) with Apple, and am - amongst other unspeakable methods - currently crossing fingers it will be picked up and acted upon 🤞

**Comment by xmddmx** [comment]
> For your case, only 2 monitors, the current 13.4 developer beta does fix the issue.

I'm still seeing occasional (like 1% to 5%) failure rate on a  2 monitor system under 13.4 beta 3, so I would clarify this to say "it's improving" but I hesitate to say it's fixed.

**Comment by glouel** [comment]
Yeah sorry, I should have expanded on that. 

I did report the small failure rate on 2 monitors as part of my aforementionned 3+ monitor bug report too, to be clear.

**Comment by nickroberts** [comment]
Just updated to 13.4 beta 4...no change. 3 monitors, only visible on a single external (and only part of the bottom of the screen).

All of the built-in screensavers seem to work across all 3 monitors (although they aren't spanned).

**Comment by glouel** [comment]
Thanks for confirming @nickroberts 

Regarding built in, it's not all, but Messages is an example of a built in one that fails (again remember the failure rate is around 80% so it anything can work occasionnally).

**Comment by glouel** [comment]
@nickroberts apologies, you are correct, I just tried with beta4 and Messages does work in beta4 unlike previous macOS builds. This is not a particularly good omen for a complete fix if they just fixed their own stuff (again) and not 3rd parties. 

(And reinstalling Aerial won't do anything at this point, this is a bug in macOS if that wasn't clear)

**Comment by DSBlackHeart** [comment]
Did this update today, Still no go.

![Screenshot 2023-05-03 at 4 24 28 AM](https://user-images.githubusercontent.com/127451632/236145977-2b315738-d1c3-4772-92c6-e9a2a4ed1fc4.jpg)

**Comment by DSBlackHeart** [comment]
So just curious is anyone after this last update having one of their monitors turn off? 

It is not disconnected from the system " If I have a web browser open on it and playing Netflix for example I still hear sound and the monitor shows in the Displays Prefs pane " 

If I restart it comes back. 

If it continues to do this I will switch the input on the back of my I-Mac to rule out a bad monitor/adapter.

**Comment by xmddmx** [comment]
Just tested in today's 13.4, no change - 2 monitors works perhaps 95% to 99% of the time.   3+ monitors fails 100% of the time.

The only noticeable change from earlier is that Apple's **Message** screensaver now works properly, whereas before it was the one Apple screensaver that also showed the bug.

**Comment by glouel** [comment]
Can confirm all the things : 
- they *only* fixed Messages in 13.4beta 4 (and now RC).
- 3+ fails 95%+ of the time   
- 2 still fails 2/5% of the time

I updated my bug reports but at this point it's clear we won't get any resolution prior to 13.5beta1 at best. 

Again, sorry to everyone affected by this. I'm as frustrated as everyone on this but I'm still confident this will get fixed by Apple.

**Comment by xmddmx** [comment]
There's a report over on MacRumors about failures using Apple's "Shifting Tiles" screensaver, and I am seeing some weirdness as well.  Basically, if you use Shifting Tiles with colors only, it works fine, but if you are using a photo album or folder of pictures, there are strange delays on 2 or 3 monitors.  

I don't know if this is the same bug, or perhpas a new bug?  However, I think the more bugs we can find which also affect Apple's own screensavers, the better.

If anyone can reproduce problems with **Shifting Tiles** it would be good to add that to your FB case (or make a new FB case?)

See https://forums.macrumors.com/threads/multimonitor-screensaver-bug-in-ventura-13-3.2384444/post-32155590

**Comment by jamesrn** [comment]
What I find strange is that this issue only occurred for me once I upgraded Aerials to 3.2.5 today. There was no change to macOS between when it was working well, and when it failed to work on my dual screen setup.
MacBook Pro M1 Dual Display.
macOS Ventura 13.3.1 (a)
If I could figure out how to revert to the previous version, whatever that was I would test.

**Comment by glouel** [comment]
> macOS Ventura 13.3.1 (a)
> If I could figure out how to revert to the previous version, whatever that was I would test.

@jamesrn you can find every version here : https://github.com/JohnCoates/Aerial/releases

Make sure you quit Companion if you are running it, close System Settings, and double click the .saver file from the zip you downloaded. 

But at this point I'm 100000% confident this issue is not linked to today's release. We got it pretty well documented, we know there's a partial fix in 13.4, and we know how partial it is. We also have good reasons to believe Apple is actilvely working on it. 

Also, know that it's not a 100% failure rate, it's more like 80/90% so that could impact your perception too. I wouldn't waste time to try older versions, but if you do, don't just test once, as it usually works the first time anyway.

**Comment by DSBlackHeart** [comment]
Morning, 

Well, I found out that my "Display Turning Off" problem was due to a faulty display, One of my 2 - 27" Thunderbolts finally bit the dust. So I replaced them both with the Apple Studio Display's .... 

Beautiful displays. Aerial still does not work ( GRRRRRRRR ) 

What I have noticed is with the recent update that when My I-Mac wakes from screen saver or sleep it acts as if it has a hang-over. Everything reacts slowly, it takes 15 to 20 seconds for it to catch up. My I-mac is no slouch, I have never seen this thing bogged down by something. 

Is anyone else seeing this?

I can't believe this is the same issue with displays & Aerial. ( We have documented this thing to death )

If it is something new is there a documented report already made? 

Thanks in advance 

Stu

![Screenshot 2023-05-18 at 6 01 12 AM](https://github.com/JohnCoates/Aerial/assets/127451632/efb6f162-e13d-4ea5-9f44-19ce853135e9)
![Screenshot 2023-05-18 at 6 04 02 AM](https://github.com/JohnCoates/Aerial/assets/127451632/d6389b14-8cf6-4ca2-9903-15182ae1d74d)

**Comment by Fofer** [comment]
> Yes, I came here to report the same with macOS 13.3. I just added a 2nd 4K Dell display to enjoy it with Ventura on my M1 Mac mini. Only once, did I see Aerial display with the video "spanning" across both screens, and now I can't get it to display that way again. I've tried everything, It's only showing up on the left screen, not my primary. The other screensavers I have installed are working fine. Sad, as Aerial is, by far, the best screensaver in the universe! Hopefully it can be addressed, otherwise I'll miss it greatly :(

Just wanted to followup, I upgraded to Ventura 13.4 today and this functionality is working just fine now on my M1 Mac mini setup.  Two Displays, one horizontal and one vertical, with the "spanning" feature including both nicely.  Super cool!

**Comment by DSBlackHeart** [comment]
Just curious what build of 13.4 are you on ?  I am using 13.4 9 (22F66) 

On both my M1 PowerBook & Intel I-Mac

**Comment by Fofer** [comment]
macOS 13.4 (22F66) on 2020 M1 Mac Mini

**Comment by JohnHales** [comment]
I just tried on the GA version of 13.4 and it still doesn’t work on my 4 monitor system.

John

From: Fofer ***@***.***>
Date: Thursday, May 18, 2023 at 18:30
To: JohnCoates/Aerial ***@***.***>
Cc: John Hales ***@***.***>, Comment ***@***.***>
Subject: Re: [JohnCoates/Aerial] Screen Savers broken function on 2+ monitor system in Ventura 13.3 beta (FB12023530) (Issue #1286)

macOS 13.4 (22F66) on 2020 M1 Mac Mini

**Comment by glouel** [comment]
To recap again, this is only (kinda) fixed in 13.4 for people with 2 monitors only (it still randomly fails 5% of the time).

Setups with 3 or more screens still fail all the time. Hopes for those are on 13.5 beta now.

**Comment by Retroghost-X** [comment]
To confirm others comments but to add a little. 

MacOS 13.4, m1 mac, three monitors, Aerial only displays on one monitor in full screen. Internal Apple screen savers all work across all monitors at once - this is different to 13.3 where Apple screensavers had issues too for me. So this is now Aerial restricted under 13.4. I am not sure 13.5 will actually fix it as it may not be a macOS issue.

**Comment by glouel** [comment]
@Retroghost-X so yes, they fixed the issue for their own 1st party screensavers (like Messages), we mentioned it above, around beta 4 if I recall correctly. As we pointed at the time, that was disappointing. 

However, nope this is not an Aerial issue, I can confirm to you that this is an issue with every single 3rd party screen saver (feel free to try any yourself).

The issue is precisely around `legacyScreenSaver.appex`, which is the macOS bridge that Apple made to run all 3rd party screen savers (and some 1st party, but they changed that to fix their issue) since Catalina. This thing has been the root cause of many many issues, and still is. I've lost countless hours to this thing, reporting bugs to Apple, working around their bugs, or just doing support every time something breaks. This is exhausting. 

Apple has a new API for their own screensavers, which we can't use yet. And even they don't use it all the time themselves (they do have other "bridges"). It's a real mess, but there's nothing I can do about it.

**Comment by Retroghost-X** [comment]
@glouel - many thanks for that reply, that is really helpful for me. Apologies for causing annoyance as a newbie. Your work and effort is really appreciate across the world. 

What you describe sounds intensely frustrating and unhelpful on Apple's part.

**Comment by glouel** [comment]
@Retroghost-X no worries, please don't take this personally, I can assure you my frustration is directed at Apple and their bug report process 😩 

I've reported this precise issue the first day of the beta 13.3 beta cycle in *february*.  Yet we're still here mid may!

**Comment by DSBlackHeart** [comment]
@glouel 
I have a curious question?  And outside the BOX

What if one were to copy the "legacyScreenSaver.appex" from say Monetry or Catalina and then insert it into Ventura ?  Would it work ?
I realize to do it you may have to disable SIP to get it done, but has the app changed so much that it would not work or are there other apps that would just cause a crash ?

**Comment by glouel** [comment]
@DSBlackHeart so as far as I understand, nowadays macOS stores all its files in a read only boot volume. This means you can't replace system files, and disabling SIP won't be enough to get around it. They are reusing what they did with iOS, the system is mounted as read only, you can't modify it, and then it's super easy to handle the upgrades/reinstall. It's pretty good for security, but it's annoying in some instances, like here. 

There's more details here : https://superuser.com/questions/1495124/read-only-file-system-when-i-try-to-touch-in-mac-os-catalina

There was a workaround early on but it doesn't seem to work anymore.

**Comment by DSBlackHeart** [comment]
Ok that is what I thought, I know to do custom ringtones you have to disable SIP & use a hack tool  to get around, it is a pain. Not for the timid, average user, you can open the system up to some pretty bad crap.

**Comment by DSBlackHeart** [comment]
Well Apple is notorious for including things "As Features" that no one will ever use.

**Comment by nickroberts** [comment]
Still broken on 13.5 Beta (22G5027e) for >= 3 monitors.

Not sure why I even bother anymore. Haha. Hopefully someday we can get a fix! I so miss my Aerial!

**Comment by xmddmx** [comment]
For those of you with 3+ monitors, please also test using Apple's **Sliding Tiles** screensaver, set to use **Photo Library** -- in our tests, this is behaving strangely on 3+ monitors, suggesting that this is indeed an Apple bug, and not something limited only to third-party screensavers.

**Comment by DSBlackHeart** [comment]
@xmddmx What am I looking for ?

**Comment by DSBlackHeart** [comment]
@xmddmx You mean Shifting Tiles or sliding panels?

**Comment by xmddmx** [comment]
My mistake, **Shifting** Tiles - 
See my prior post https://github.com/JohnCoates/Aerial/issues/1286#issuecomment-1542332240

**Comment by DSBlackHeart** [comment]
@xmddmx Thanks I will set it up and try and break it

**Comment by xmddmx** [comment]
Testing with 13.5 beta on 2 monitors, and still seeing the "5% failure" mode: 95% of the time it works fine, but 5% the original bug shows up: screen is 1 blank, screen 1's content on screen 2.

**Comment by DSBlackHeart** [comment]
@xmddmx      CONFIRMED i tested it with a library that has 30K photos .... Its Broke

**Comment by glouel** [comment]
I also get slight delays here but not as bad as other reports, though I do have a pretty small library on the mac I used to test it, so that could be it. This is also interesting because the "picture" screensavers from Apple have a special appex, they use `iLifeSlideshows.appex`(I love the throwback in the naming, considering it was "only" introduced in Catalina !) instead of `legacyScreenSaver.appex` which has lots of extra entitlements we don't get. 

Anyway, it got me curious into how the *heck* they fixed Messages. I was under the impression that it was still a .saver, as Apple basically never use the new `.appex` API they built and last year's Monterey saver, for example, was released as a `.saver`. For the curious, those are stored in `/System/Library/Screen Savers`. 

So I tried Messages again with Console open and was utterly surprised to see it was running `Computer Name.appex` which I didn't expect (or maybe I forgot !). Perhaps more interestingly, the Ventura screen saver is also an `.appex`, and that's definitely new.  

For the curious they are stored in : `/System/Library/Frameworks/ScreenSaver.framework/PlugIns/`

This is what we have in all 13.x version (I went back and installed a 13.2 vm to be sure) : 
- Computer Name.appex <- That's messages		
- iLifeSlideshows.appex <- That's the bridge that is used by the picture savers I talked about above
- legacyScreenSaver.appex <- I believe this runs native arm savers on arm macs 
- legacyScreenSaver-x86_64.appex <- I believe this runs x86 savers on arm macs 
- Ventura.appex <- the new Ventura saver

So at the very least, unlike what I wrote a bit earlier above, it seems Apple is actually starting to use their new API. And that Computer Name/Messages had the exact same issue that legacyScreenSaver had, and they managed to fix `that` one. Are we closer to anything ? Nope, but maybe someday Apple will open up the `.appex` API. 

At least seeing them start using it shows movement, which is news to me.

**Comment by xmddmx** [comment]
Doing a little archaeology, I see that as of 10.14 (macOS Mojave) the Message screensaver was called **Floating Message.saver** and lived in /System/Library/Screen Savers/   

I wonder at which macOS version it changed to an .appex ?

**Comment by glouel** [comment]
Interesting that they changed the name to Computer Name (which is just the default), bit weird, old name was better (I mean I assume you can still set a custom message, didn't bother checking).

I seem to remember there was one in Catalina as an .appex but not really sure. I'll  put a few restore images overnight to download as it got me too curious 😅

**Comment by vinhtq115** [comment]
On 13.4, so far, I haven't encountered the "1 monitor working and 1 showing black screen" but now, video can be misaligned between two monitors. Previously, on 13.3, when Aerial worked with 2 monitors, it would aligned the video correctly. This is an example of it (mine is set to Cloned and Fit to screen):

![IMG_7343](https://github.com/JohnCoates/Aerial/assets/34190240/dbdc20d0-f1e1-49d1-9cac-6d9c50a2bbb1)

**Comment by glouel** [comment]
@vinhtq115 that's intriguing with a 2 monitor setup. Does it happen 5% of the time roughly, or every time ? 

If you can please send me a log (go into Advanced, enable debug mode, launch it once, then grab the log using the show in finder button). Please note the exact local time you saw it occuring too so I can check.

When it bugs for most people, you get the main screen shown on the secondary one, which is why you get the mismatch in size. It's surprising though to have it play correctly on 1st. Hadn't seen that one so very curious about the log. Al

**Comment by vinhtq115** [comment]
> @vinhtq115 that's intriguing with a 2 monitor setup. Does it happen 5% of the time roughly, or every time ?
> 
> If you can please send me a log (go into Advanced, enable debug mode, launch it once, then grab the log using the show in finder button). Please note the exact local time you saw it occuring too so I can check.
> 
> When it bugs for most people, you get the main screen shown on the secondary one, which is why you get the mismatch in size. It's surprising though to have it play correctly on 1st. Hadn't seen that one so very curious about the log.

That's the first time I have seen that bug so I think that it is uncommon. I will send you the log when it happens again.

**Comment by vinhtq115** [comment]
Here's the log. It happened somewhere around 10:51.

[AerialLog.txt.zip](https://github.com/JohnCoates/Aerial/files/11523681/AerialLog.txt.zip)

![IMG_7344](https://github.com/JohnCoates/Aerial/assets/34190240/34998719-ee01-472c-9b37-91ec23f6f83d)

**Comment by glouel** [comment]
Thanks @vinhtq115 

Looks like you have a variation of the bug, because possibly of using clone mode. But it's the same root 5% issue with 2 monitors that we were talking about. 

I believe looking at your right screen that even that one isn't correct (black bars at the bottom, not on top) so it's likely more mismatch. Everything is reported correctly in your logs just like in other cases. 

Let's hope this get fixed quickly

**Comment by LeakedDave** [comment]
Just hear to drop sad face :(

My screens have been black for a week and I was wondering why... lmao GG

**Comment by GabeRoze** [comment]
Been bothering me a while:


Mac OS 13.4 (22F66)
Apple M1 Max 2021

No matter the settings, I see a screensaver on my vertical monitor (only on the bottom half)
If I set the rotation to 0 on the vertical monitor, than the screensaver only shows on that monitor.



3 Monitors
![image](https://github.com/JohnCoates/Aerial/assets/780654/ec05ea63-fe49-404c-b540-55ade9bff5c9)

**Comment by eduzaka** [comment]
> Been bothering me a while:
> 
> Mac OS 13.4 (22F66) Apple M1 Max 2021
> 
> No matter the settings, I see a screensaver on my vertical monitor (only on the bottom half) If I set the rotation to 0 on the vertical monitor, than the screensaver only shows on that monitor.
> 
> 3 Monitors ![image](https://user-images.githubusercontent.com/780654/240479510-ec05ea63-fe49-404c-b540-55ade9bff5c9.png)

Same configuration, same problem here.

**Comment by theeddie** [comment]
Im having the same issues, ive tried HDR none HDR.  about 10% of the screen savers actually work

**Comment by glouel** [comment]
Again, my weekly recap, the bug is with macOS and all 3rd party screensavers, we're waiting for Apple to fix the issue. 

- 3+ monitors will work around 5% of the time and fail otherwise. 
- 2 monitors will work around 95% of the time and fail otherwise (* for most people, I've had one report on discord of things being more wrong for someone, we don't know why)

13.2.X doesn't exhibit the bug.
13.4 has the results above. 13.5beta1 doesn't improve things. 

Hopes on beta2. I understand it's an annoying bug for a lot of people, I feel your pain, but there's nothing more I can do at the moment despite my best efforts, this is on Apple.

**Comment by DSBlackHeart** [comment]
System Version:	macOS 13.5 (22G5038d)

Still not working, Why do I get the feeling Apple will release a new OS before they fix this...

**Comment by glouel** [comment]
> System Version: macOS 13.5 (22G5038d)
> 
> Still not working, Why do I get the feeling Apple will release a new OS before they fix this...

At this point that's a pretty safe bet. 

It's also my yearly prediction of "*surely* this year Apple will introduce the new Screen Saver SDK using .appex"

So keep an eye out on monday, this may be the most important Apple news of wwdc, at least for me  ;)

**Comment by Turbotailz** [comment]
Updating to 13.4 fixed this problem for me when using 2 screens, but when I use 3 the screensaver only appears on 1.

**Comment by panosru** [comment]
Oh ¢r@p! If 13.4 fixed the use of screens but not 3+, I guess I'll have to wait for macOS v16 with 7 monitors 🤣

**Comment by asportnoy** [comment]
Yep, Aerial built into macOS Sonoma
![Screenshot 2023-06-05 at 12 46 02](https://github.com/JohnCoates/Aerial/assets/14863373/7944b096-4cc5-40da-b8f7-2f1317d69fa6)

**Comment by glouel** [comment]
So that wasn't the screen saver news I was hoping for 😅

I'm waiting to install Sonoma to see what they did with their version, it will be interesting to see what they are doing and if it justifies continuing development of Aerial.

**Comment by xmddmx** [comment]
Downloading 14 beta now - would it make sense to start some new threads for Sonoma issues?

**Comment by glouel** [comment]
Yep please report anything Sonoma related in a new issue, unless it's this specific one I guess.

**Comment by xmddmx** [comment]
See https://github.com/JohnCoates/Aerial/issues/1305

**Comment by xmddmx** [comment]
Has anyone tested 13.5 beta 2 (which was released 31 May 2023) ?   

Looks like I tested [13.5 beta 1](https://github.com/JohnCoates/Aerial/issues/1286#issuecomment-1555923878), but missed beta 2, and now that I'm testing 14.0 beta, I can't seem to go backwards.

**Comment by DSBlackHeart** [comment]
I am using :
Version 13.5 Beta (22G5038d)
Still does not work, I am able to load Sonoma on my MBP but I dont see it yet if ever for my Intel I-Mac.

**Comment by glouel** [comment]
I took the opportunity to talk to DTS at Apple about the issue tonight, here's the latest update I can share : 

- engineering is aware of the issue, it's an issue on their end
- because they are changing things (you know...), and it's wwdc time, they can't promise a resolution date
- they are committed to fixing the issue

Last thing, if you are affected as a user of Aerial, I was told it was a good idea to contact https://getsupport.apple.com as this can help bugs get prioritised. So please consider doing so. If you do so, I would recommend mentioning that you have the issue with multiple 3rd party screen savers.

**Comment by glouel** [comment]
No change in 13.5beta4 FYI.

**Comment by brianburkett** [comment]
Just chiming in that this is a major issue for OLED displays.

I run a triple monitor setup, and just noticed for the first time yesterday that this bug is causing burn-in on my OLED panel.

I run 3x 4k monitors.
The main monitor is an OLED display.

Every single time, the screensaver only shows in lower lefthand corner on the OLED screen (covering ~60% of the screen), while the other 2 displays are blank.

This bug now has caused a noticeable burn-in on the main monitor (fortunately fixed via a pixel refreshing tool on the monitor) — but if you have an OLED, be weary of this issue until it's resolved to avoid an expensive problem.

**Comment by glouel** [comment]
Can anyone test 13.5beta5 and get back to us ? Based on some activity on my radars, I believe the fix may have been implemented in that beta, but would need confirmation. Thanks a lot !

**Comment by royvella** [comment]
Not seeng beta5 to give it a go… only beta2 actually.

**Comment by glouel** [comment]
![Capture d’écran 2023-07-10 à 19 38 05](https://github.com/JohnCoates/Aerial/assets/37544189/4d6f916f-6d6b-4e49-80c4-15e8ee116b7f)

I sent my message early, it may take an hour to be available, until it propagates through their CDNs, but yeah it's beta5. 

In any case, you should definitely **NOT** be stuck on beta 2 !

**Comment by spawnofbill** [comment]
> Can anyone test 13.5beta5 and get back to us ? Based on some activity on my radars, I believe the fix may have been implemented in that beta, but would need confirmation. Thanks a lot !

Downloading now on my secondary machine, will let you know.

**Comment by zeromhz** [comment]
> > Can anyone test 13.5beta5 and get back to us ? Based on some activity on my radars, I believe the fix may have been implemented in that beta, but would need confirmation. Thanks a lot !
> 
> Downloading now on my secondary machine, will let you know.

The issue appears to be fixed on 13.5beta5!

**Comment by glouel** [comment]
> The issue appears to be fixed on 13.5beta5!

A well deserved, FINALLY!

This has been a few hectic months, I can't wait for 13.5 to go final so that I can finally stop supporting twice a day to people about that issue 🎉

**Comment by DSBlackHeart** [comment]
Great news :) , Beta 2 was mentioned earlier and destroyed my I-Mac, I am currently back in Monterey. I was waiting to hear we have something stable. Thanks

**Comment by gman-wa** [comment]
> > > Can anyone test 13.5beta5 and get back to us ? Based on some activity on my radars, I believe the fix may have been implemented in that beta, but would need confirmation. Thanks a lot !
> > 
> > 
> > Downloading now on my secondary machine, will let you know.
> 
> The issue appears to be fixed on 13.5beta5!

Confirmed - fixed for me on 13.5beta5.  Also other screensavers like Padbury Clock.

**Comment by glouel** [comment]
@DSBlackHeart 

GM : I would expect soon, it's likely the last beta in this cycle. 

Sonoma : Totally different beast, they broke/rewrote everything, see the other thread for more details : https://github.com/JohnCoates/Aerial/issues/1305

It's likely they broke Ventura in preparation for Sonoma. But Sonoma breaks things so so so much more profoundly, it's a whole different situation. It's early though.

**Comment by DSBlackHeart** [comment]
Kinda what I thought, thanks for the quick reply, I was curious because Sonoma destroyed my I-Mac so I am back on Monterey for now, I was waiting for a Stable Ventura or Sonoma

Awesome to hear though

**Comment by patrickschaper** [comment]
OMG... 13.5 actually fixes the problem 🥳

**Comment by glouel** [comment]
I kinda missed it was released yesterday, I'll finally close this and keep it pinned so people can find it.

Again everyone, thanks for your patience and understanding, this was not a fun last few months 😩

**Comment by jiclark** [comment]
Thanks Guillaume! Hang in there!!On Jul 25, 2023, at 7:06 AM, Guillaume Louel ***@***.***> wrote:﻿
Closed #1286 as completed.

**Comment by JohnHales** [comment]
So happy this is finally resolved! Thanks for all who pushed it with Apple!

John

Sent from John's iPad. Please excuse any typos.
________________________________

**Comment by Retroghost-X** [comment]
Well done for sticking with it! 

I confirm fixed on 13.5 on all my Macs, multiscreen beauty returns once again!

Great stuff...

**Comment by eduzaka** [comment]
Yes! Workig again! 🔝

**Comment by RoyRao2333** [comment]
Well, I don't want to spoil the fun... But it  still occasionally happens on 13.5...

![IMG_4569](https://github.com/JohnCoates/Aerial/assets/31413093/b9f11530-a9cd-45fb-9944-eecd72dfa764)

<img width="609" alt="image" src="https://github.com/JohnCoates/Aerial/assets/31413093/34843d98-0fd1-4b43-9aa5-171a700ea0e0">

<img width="708" alt="image" src="https://github.com/JohnCoates/Aerial/assets/31413093/ce2c299e-79b0-4c87-8c7b-686acfe1380c">

**Comment by glouel** [comment]
> Well, I don't want to spoil the fun... But it still occasionally happens on 13.5...

Do you have an idea of roughly how often ? Please use Feedback Assistant to report this to Apple so they can check if there' some bug specific to your configuration or if it's something else.

**Comment by RoyRao2333** [comment]
@glouel 

It happened twice or so since 13.5 stable is released. I'll continue my observation of this and send a feedback to Apple if needs to. Thanks for the reply.

**Comment by glouel** [comment]
> It happened twice or so since 13.5 stable is released. I'll continue my observation of this and send a feedback to Apple if needs to. Thanks for the reply.

@RoyRao2333 I'm not 100% surprised about this, many aspects of this bug over the months look like a race condition so it's quite probable that it still sometimes fail. Sonoma changes everything *again* with the ability to set a different screen saver per screen/space, with different issues, so I wouldn't be surprised if whatever is the situation in 13.5 gets set in stone for the 13.X line. Sonoma is the new battle front 😅

**Comment by RoyRao2333** [comment]
@glouel 

And it just happened again yesterday 🙄 Losing faith in Apple gradually...
